katdal/h5datav3.py

```diff
@@ -112,6 +112,10 @@ def dummy_dataset(name, shape, dtype, value):
     return dummy_file.create_dataset(name, shape=shape, maxshape=shape,
                                      dtype=dtype, fillvalue=value, compression='gzip')
 
+
+class AttributeFound(Exception):
+    """This indicates that an attribute has been found and contains its value."""
```

Comment:
This feels like it should be an anti-pattern (using exceptions for non-exceptional things), but I can see that it makes the code simpler and I don't have a better solution, so carry on. Maybe with Python 3.3+ it will be possible to do something clever with generators and `yield from`.

Reply:
I like your Python 3 suggestion for the future.